src/pages/Popup/PopupPanel.tsx

```diff
@@ -1,87 +1,103 @@
+import { logAddressSearchClick, logUrlSearchClick } from '../../shared/logs';
 
 import '../../shared/reset.css';
 import '../../shared/font.css';
 
-import { logAddressSearchClick, logUrlSearchClick } from '../../shared/logs';
-
 interface PopupPanelProps {
     hideAlpha?: boolean;
     hideSettings?: boolean;
-    standalone?: boolean;
 }
 
-export function PopupPanel({ hideAlpha = false, hideSettings = false, standalone = false }: PopupPanelProps) {
-    const [to, setTo] = useState('');
-    const [chainId, setChainId] = useState('0x1');
-    const [url, setURL] = useState('');
+const emptyState = {
+    mode: SearchMode.Address,
+    chainId: '0x1',
+    url: '',
+    address: '',
+};
+
+export function PopupPanel({ hideAlpha = false, hideSettings = false }: PopupPanelProps) {
+    const [compactMode, setCompactMode] = useState(false);
+    const [searchInput, setSearchInput] = useState<SearchState>(emptyState);
 
     const descriptionResult = useAsyncCallback(async (chainId, to) => fetchDescription(chainId, to));
     const analyzeResult = useAsyncCallback(async (chainId, to, url) => fetchAnalyze(chainId, to, url));
 
     async function submitUrl(url: string) {
-        setURL(url);
+        setCompactMode(true);
         analyzeResult.execute(undefined, undefined, url);
         logUrlSearchClick(url);
     }
 
     async function submitAddress(chainId: string, to: string) {
-        setChainId(chainId || '');
-        setTo(to || '');
+        setCompactMode(true);
         descriptionResult.execute(chainId, to);
         analyzeResult.execute(chainId, to, undefined);
         logAddressSearchClick(to, chainId);
     }
 
-    const compact = to !== '' || url !== '';
-
-    return (
-        <ErrorBoundary>
-            {!hideAlpha && <Layout.Banner>BETA</Layout.Banner>}
+    async function handleSubmit(event: React.SyntheticEvent<HTMLFormElement>) {
+        event.preventDefault();
+        if (searchInput.mode === SearchMode.Address) {
+            await submitAddress(searchInput.chainId, searchInput.address);
+        } else {
+            await submitUrl(searchInput.url);
+        }
+    }
 
-            <Layout.Logo hidden={compact}>
-                <Logo />
-                <div>Blockfence</div>
-            </Layout.Logo>
+    function reset() {
+        setSearchInput(emptyState);
+        setCompactMode(false);
+        analyzeResult.reset();
+        descriptionResult.reset();
+    }
 
-            <Layout.FloatingSettings>{!hideSettings && !compact && <SettingsMenu />}</Layout.FloatingSettings>
+    const isLoading = analyzeResult.loading || descriptionResult.loading;
 
-            <Layout.Header severity={analyzeResult.result?.severity}>
-                <SearchBar
-                    onAddressClick={submitAddress}
-                    onURLClick={submitUrl}
-                    disabled={analyzeResult.loading || descriptionResult.loading}
+    return (
+        <ErrorBoundary>
+            <form onSubmit={handleSubmit}>
```

Comment:
QQ: does the <form> DOM element has any  effect on the UI or is it purely funcional?